### ISSUE TITLE
Parsing Error

### ISSUE THREAD

**Issue by Sono-Gomorrha** [Tzahi12345/YoutubeDL-Material#261]
I installed version 4.1 on an Ubuntu 20.04.1 LXC system (under Proxmox) following the instructions. Output of lsb_release -a:
```
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.1 LTS
Release:        20.04
Codename:       focal

```
When I enter a URL (I've tested with three different ones and also with video and audio only) I immediately get a 'Download failed!' error. 

This is the output in the Linux shell:

```
2020-11-23T09:31:45.998Z ERROR: Error during parsing:Error: Command failed with exit code 127: /root/youtubedl-material/node_modules/youtube-dl/bin/youtube-dl -i --dump-json https://youtu.be/ZsjK4VDopiE
2020-11-23T09:31:46.003Z ERROR: failed to parse for urls starting with https://youtu.be/ZsjK4VDopiE
(node:26986) DeprecationWarning: uuidv4() is deprecated. Use v4() from the uuid module instead.
2020-11-23T09:31:47.785Z ERROR: Command failed with exit code 127: /root/youtubedl-material/node_modules/youtube-dl/bin/youtube-dl -i --dump-json -f best -o video/%(title)s.mp4 --write-info-json --print-json -f best[ext=mp4] https://youtu.be/ZsjK4VDopiE
```
And this is the Output in Chrome -> inspect -> console
```
polyfills-es2015.5b408f108bcea938a7e2.js:1 POST http://192.168.2.60:17442/api/tomp4?apiKey=4241b401-7236-493e-92b5-b72696b9d853&sessionID=fe7666469fe8a1d7d31a3e87fbfa8335 500 (Internal Server Error)
g @ polyfills-es2015.5b408f108bcea938a7e2.js:1
scheduleTask @ polyfills-es2015.5b408f108bcea938a7e2.js:1
onScheduleTask @ polyfills-es2015.5b408f108bcea938a7e2.js:1
scheduleTask @ polyfills-es2015.5b408f108bcea938a7e2.js:1
scheduleTask @ polyfills-es2015.5b408f108bcea938a7e2.js:1
scheduleMacroTask @ polyfills-es2015.5b408f108bcea938a7e2.js:1
l @ polyfills-es2015.5b408f108bcea938a7e2.js:1
(anonymous) @ polyfills-es2015.5b408f108bcea938a7e2.js:1
P.o.<computed> @ polyfills-es2015.5b408f108bcea938a7e2.js:1
(anonymous) @ 1-es2015.1aeca98d815b032fb70e.js:1
_trySubscribe @ main-es2015.0cbc545a4a3bee376826.js:1
subscribe @ main-es2015.0cbc545a4a3bee376826.js:1
i @ main-es2015.0cbc545a4a3bee376826.js:1
_innerSub @ main-es2015.0cbc545a4a3bee376826.js:1
_tryNext @ main-es2015.0cbc545a4a3bee376826.js:1
_next @ main-es2015.0cbc545a4a3bee376826.js:1
next @ main-es2015.0cbc545a4a3bee376826.js:1
(anonymous) @ main-es2015.0cbc545a4a3bee376826.js:1
_trySubscribe @ main-es2015.0cbc545a4a3bee376826.js:1
subscribe @ main-es2015.0cbc545a4a3bee376826.js:1
call @ main-es2015.0cbc545a4a3bee376826.js:1
subscribe @ main-es2015.0cbc545a4a3bee376826.js:1
call @ 1-es2015.1aeca98d815b032fb70e.js:1
subscribe @ main-es2015.0cbc545a4a3bee376826.js:1
call @ main-es2015.0cbc545a4a3bee376826.js:1
subscribe @ main-es2015.0cbc545a4a3bee376826.js:1
downloadClicked @ 1-es2015.1aeca98d815b032fb70e.js:1
(anonymous) @ 1-es2015.1aeca98d815b032fb70e.js:1
Bi @ main-es2015.0cbc545a4a3bee376826.js:1
o @ main-es2015.0cbc545a4a3bee376826.js:1
(anonymous) @ main-es2015.0cbc545a4a3bee376826.js:1
invokeTask @ polyfills-es2015.5b408f108bcea938a7e2.js:1
onInvokeTask @ main-es2015.0cbc545a4a3bee376826.js:1
invokeTask @ polyfills-es2015.5b408f108bcea938a7e2.js:1
runTask @ polyfills-es2015.5b408f108bcea938a7e2.js:1
invokeTask @ polyfills-es2015.5b408f108bcea938a7e2.js:1
h @ polyfills-es2015.5b408f108bcea938a7e2.js:1
p @ polyfills-es2015.5b408f108bcea938a7e2.js:1
Show 8 more frames

```

I'm kinda lost here as to finding the origin of the error. When I simply run youtube-dl and the video url in the linux shell it works and I get an mkv file in the end. So the youtube-dl part itself works. It looks like youtubedl-material does not succeed in handing over the command to youtube-dl.

When I try to directly run youtubedl-material/node_modules/youtube-dl/bin/youtube-dl I get the '/usr/bin/env: 'python': No such file or directory' error. This looks like the culprit. 
Running which python gives me no output. Running which python3 gives me /usr/bin/python3

**Comment by Tzahi12345** [comment]
Looks like your local youtube-dl is horribly outdated, however the server has its own version which I think takes precedence, but don't quote me on that. You can find the local version by opening up the `details` file in the youtube-dl node_modules folder you cleverly found.

You may need to rename Python, but it sounds like an ugly approach. Or, if this is possible, simply make an alias of `python3` called `python` ([found these instructions](https://askubuntu.com/a/610079)). Seems like the lower cost solution, so I would try the alias method first. Let me know if either works!

Also this got me thinking that we should probably check for these dependencies on startup to make it clear that they are missing.

**Comment by Sono-Gomorrha** [comment]
You are right about the version. The outdated youtube-dl is the latest official package from the debian/Ubuntu repo. Normally I would rather take the latest version directly from github, but as I had some issues with youtubedl-material I thought I first follow the instructions to rule out my deviation is reason for my error.

Yes, I have checked the [details](url) file and it is referring to and up to date release.
`{"version":"2020.11.21.1","path":null,"exec":"youtube-dl"}`

I've put an alias alias `python='python3'` into .bashrc (another approach I use on other systems as well) and now when I enter python I directly get the python 3.8.5 console. However, trying to run the binary from the folder above still gives the same error. 

I installed python from the repo (python 2.7.18) via normal package manager, now when trying to run the youtube-dl from `/youtubedl-material/node_modules/youtube-dl/bin/` I no longer get that error.

In the web interface I can now also download videos and audio as well. So it was the python issue. I would like to recommend that you check which version of python is being installed on a system to prevent this. 
Also I'm not sure if the instructions are confusing or if I'm lacking something here: Prerequisites says "sudo apt-get install nodejs youtube-dl ffmpeg", but is a manual install of youtube-dl actually necessary when the server uses it's own version?

Note: When downloading something through the web-interface I get this error in the linux shell:
`(node:372) DeprecationWarning: uuidv4() is deprecated. Use v4() from the uuid module instead`
